### ISSUE TITLE
RoaringBitmap.andCardinality is not reflexive with run containers

### ISSUE THREAD

**Issue by openkappa** [RoaringBitmap/RoaringBitmap#153]
RoaringBitmap.andCardinality should return the cardinality of the bitmap when the two bitmaps are identical. RoaringBitmap.equals respects this relation, but RoaringBitmap.andCardinality fails when there are run containers in the bitmap.

The second assertion below fails, whereas the first does not.

   ```
@Test
    public void testAndCardinality() {
        RoaringBitmap baseline = new RoaringBitmap();
        baseline.add((1L << 16) + 1, 1L << 18);
        baseline.add((1L << 19) + 1, 1L << 20);
        baseline.add((1 << 21) + 1);
        baseline.add((1 << 21) + 3);
        baseline.add((1 << 21) + 5);
        assertEquals(baseline, RoaringBitmap.and(baseline, baseline));
        assertEquals(baseline.getCardinality(), RoaringBitmap.andCardinality(baseline, baseline));
    }
```

java.lang.AssertionError: 
Expected :720897
Actual   :1

**Comment by gssiyankai** [comment]
@openkappa
Thanks for reporting the issue. Fixed with #154

**Comment by lemire** [comment]
Fixed in v0.6.40
